app/migrations/Version20231205094436.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Mautic\Migrations;
+
+use Doctrine\DBAL\Schema\Schema;
+use Mautic\CoreBundle\Doctrine\PreUpAssertionMigration;
+use Mautic\LeadBundle\Entity\ListLead;
+
+final class Version20231205094436 extends PreUpAssertionMigration
+{
+    protected function preUpAssertions(): void
+    {
+        $this->skipAssertion(function (Schema $schema) {
+            return $schema->getTable($this->getTableName())->hasIndex($this->getIndexName());
+        }, sprintf('Index %s already exists', $this->getIndexName()));
+    }
+
+    public function up(Schema $schema): void
+    {
+        $query = sprintf(
+            'ALTER TABLE %s ADD INDEX %s (lead_id, leadlist_id, manually_removed)',
+            $this->getTableName(),
+            $this->getIndexName()
+        );
```

Comment:
@aarohiprasad I tried to execute the migration on my local machine, but no changes in DB, Please see my following suggestion:
```suggestion
        $this->addSql(sprintf(
            'ALTER TABLE %s ADD INDEX %s (lead_id, leadlist_id, manually_removed)',
            $this->getTableName(),
            $this->getIndexName()
        ));
```